[New Term Entry]: Java String `.format()` method (#809)
Codecademy/docs: content/java/concepts/strings/terms/format/format.md

---
Title: '.format()'

> Description: 'The .format() method returns a string with additional arguments in a specifically defined format'

Author: ```suggestion
Description: 'Returns a string with additional arguments in a specifically defined format'
```

---
Title: '.format()'
Description: 'Returns a string with additional arguments in a specifically defined format.'
Subjects:
  - 'Computer Science'
Tags:
  - 'Strings'
  - 'Methods'
CatalogContent:
  - 'learn-java'
  - 'paths/computer-science'
---

> The `.format()` method returns a string with additional arguments in a specifically defined format.

Reviewer: ```suggestion
The **`.format()`** method returns a string with additional arguments in a specifically defined format.
```

## Syntax

```pseudo

> string.format(String str, ...args)

Reviewer: ```suggestion
string.format(String str, ...args);
```

```

- `str` is of type `String` and includes a **format specifier**.

> - `args` is an optional argument which can be of any data type to be returned with the first string argument.

Reviewer 2: ```suggestion
- `args` is an optional argument, which can be of any data type, to be returned with the first string argument.
```

> ## Format Specifiers

Reviewer: ```suggestion
### Format Specifiers
```

Wondering if this should be a lv. 3 header since this sort of falls under "syntax"?

Some common format specifiers include:

| Format Specifier | Type     |

> | ---------------- | -------- |

Reviewer: ```suggestion
| :----------------: | :--------: |
```

| `%b`             | boolean  |
| `%c`             | char     |
| `%d`             | integer  |
| `%f`             | float    |
| `%s`             | string   |

> **Note:** `NullPointerException` will be thrown if the format is `null`. An `IllegalFormatException` will be thrown if the specifier and argument don't match or the specifier does not exist.

Author: ```suggestion
**Note:** `NullPointerException` will be thrown if the format is `null`. An `IllegalFormatException` will be thrown if the specifier and argument do not match or the specifier does not exist.
```
